### DISCUSSION TITLE
MeshLab 2021.07 released!

### DISCUSSION THREAD

**Discussion by alemuntoni** [cnr-isti-vclab/meshlab#1041] [General]
MeshLab 2021.07 is out!

In this version we introduce several new plugins, along with lot of bug fixes and optimizations.

- New mesh booleans plugin, which replaces the old CSG plugin. The new plugin is based on the paper "[Mesh arrangements for solid geometry](https://dl.acm.org/doi/10.1145/2897824.2925901)" (implementation based on [libigl](https://libigl.github.io/));
- New plugin for importing GLTF/GLB files, based on [tinygltf](https://github.com/syoyo/tinygltf);
- New plugin for importing/exporting e57 files, based on [libe57](http://www.libe57.org/);
- New plugin for exporting NXS/NXZ files, based on [nexus](http://vcg.isti.cnr.it/nexus/).

Feel free to ask any question on this thread!

Enjoy MeshLab :)

**Comment by nicopeis** [comment]
Hi all
First of all thank you for this awesome program!
I have notticed a little issue in the las 2021.07 both d and not d version
When exporting to STL, the "binary" option is checked and cannot be unchecked. This was working fine on 2021.05. I tried it on a couple of PC and both have the same issue
Your are the best!! 🙌🙌🙌🙌

**Reply by alemuntoni** [reply]
Thank you very much @nicopeis :)
I'm aware of this bug, I'll try to solve it as soon as possible. It affects all the filter dialogs having a check box as first parameter..
The checkbox is actually checkable, but you need to click on the right part of the text of the checkbox, like here:

![image](https://user-images.githubusercontent.com/6656168/126862614-db68674c-8a58-4f6f-877d-d7c602af18d7.png)

I know it is not great, but at least every option can be checked/unchecked :)

**Comment by bednarhonza** [comment]
Hi all,
thank you for this amazing update!

Do you plan to include the GLB format for export?

**Reply by alemuntoni** [reply]
See #441.
We plan to add the exporter, but I don't know when.

**Comment by npolys** [comment]
Thanks!

The Material and Texture exporters for X3D are broken !! :-[


On Mon, Jul 19, 2021 at 12:46 PM Alessandro Muntoni <
***@***.***> wrote:

> MeshLab 2021.07 is out!
>
> In this version we introduce several new plugins, along with lot of bug
> fixes.
>
>    - New mesh booleans plugin, which replaces the old CSG plugin. The new
>    plugin is based on the paper "Mesh arrangements for solid geometry
>    <https://dl.acm.org/doi/10.1145/2897824.2925901>" (implementation
>    based on libigl <https://libigl.github.io/>);
>    - New plugin for importing GLTF/GLB files, based on tinygltf
>    <https://github.com/syoyo/tinygltf>;
>    - New plugin for importing/exporting e57 files, based on libe57
>    <http://www.libe57.org/>;
>    - New plugin for exporting NXS/NXZ files, based on nexus
>    <http://vcg.isti.cnr.it/nexus/>.
>
> Feel free to ask any question on this thread!
>
> Enjoy MeshLab :)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cnr-isti-vclab/meshlab/discussions/1041>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB2TSMYSND7WXVLLI3CQKY3TYRJGVANCNFSM5AUCI7HQ>
> .
>


-- 
Nicholas F. Polys, Ph.D.

Director of Visual Computing
Virginia Tech Research Computing

Affiliate Professor
Virginia Tech Department of Computer Science

**Reply by alemuntoni** [reply]
Hi, I checked and it seems to me that it is actually the importer that is not working with textured meshes. 
Could you please confirm this?

**Comment by ggaspard54** [comment]
Hello,

i'm trying to export to .e57 a .ply file but Meshlab crash...

My .ply file have 7M points so i decimate to less points (250k) but i still have this problem..

Suggestions ? MeshLab bug ?

thx ! :)

**Reply by alemuntoni** [reply]
If it crashes, it's a bug!
Please open an issue [here](https://github.com/cnr-isti-vclab/meshlab/issues), describing which platform are you using and sharing a mesh that causes the crash (the decimated one is better). Then I'll try to solve the issue ASAP :)

**Comment by npolys** [comment]
I believe the X3D exporter in Meshlab is broken..

 there is a bug or two about this reported in the system, but it has not
been addressed

 Materials and Textures are not output :-[


with best regards,
 _nicholas


On Wed, Aug 18, 2021 at 4:26 AM Alessandro Muntoni ***@***.***>
wrote:

> Hi, I checked and it seems to me that it is actually the importer that is
> not working with textured meshes.
> Could you please confirm this?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/cnr-isti-vclab/meshlab/discussions/1041#discussioncomment-1199392>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB2TSMYND32ADV2KSGV3UCTT5NVDLANCNFSM5AUCI7HQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
Nicholas F. Polys, Ph.D.

Director of Visual Computing
Virginia Tech Research Computing

Affiliate Professor
Virginia Tech Department of Computer Science

**Reply by alemuntoni** [reply]
@npolys I already replied about the x3d and opened this issue: #1061.
I think it's the importer more than the exporter that is broken: an x3d exported with 2021.07 and then imported with an old version of meshlab works fine. 

P.S. note that you are probably missing all the conversation and links if you reply via email. I suggest to look the conversation from the github website.
